src/goto-symex/renaming_level.h

```diff
@@ -84,6 +86,74 @@ struct symex_level2t : public symex_renaming_levelt
 {
   symex_level2t() = default;
   ~symex_level2t() override = default;
+
+  std::shared_ptr<current_namest> global_names;
+
+  /// Ages the generation of an ssa expression by one.
+  void increase_generation(const irep_idt l1_identifier, const ssa_exprt &lhs)
+  {
+    current_names.emplace(l1_identifier, std::make_pair(lhs, 0));
+    if(global_names != nullptr)
+      global_names->emplace(l1_identifier, std::make_pair(lhs, 0));
+
+    increase_generation(l1_identifier);
+  }
+
+  /// Ages the generation of an ssa expression by one. Does nothing
```

Comment:
How about "Allocates a fresh L2 name for the given L1 identifier, and makes it the latest generation on this path"